contrib/src/main/java/com/datatorrent/contrib/enrich/FixedWidthFSLoader.java

```diff
@@ -0,0 +1,248 @@
+   */
+  private transient int[] width;
+  /**
+   * Padding character. Default is white space.
+   */
+  private transient char padding = ' ';
+  private transient FixedWidthParser fixedWidthParser;
+  private transient boolean initialized;
+  /**
+   * Map holding the format for date fields
+   */
+  private transient Map<String, String> fieldDateFormat = Maps.newHashMap();
+
+  private static final Logger logger = LoggerFactory.getLogger(FixedWidthFSLoader.class);
+
+  public FixedWidthFSLoader()
+  {
+  }
+
+  /**
+   * Gets the option if file has header or not.
+   *
+   * @return hasHeader
+   */
+  public boolean isHasHeader()
+  {
+    return hasHeader;
+  }
+
+  /**
+   * Set to true if file has header
+   * 
+   * @param hasHeader
+   */
+  public void setHasHeader(boolean hasHeader)
+  {
+    this.hasHeader = hasHeader;
+  }
+
+  /**
+   * Gets the field description
+   * 
+   * @return fieldDescription
+   */
+  public String getFieldDescription()
+  {
+    return fieldDescription;
+  }
+
+  /**
+   * Sets fieldDescription i.e a String specifying information related to fields
+   * in fixed-width file. Format is [NAME]:[FIELD_TYPE]:[WIDTH]:[date format if
+   * FIELD_TYPE is DATE] FIELD_TYPE can take on of the values of
+   * {@link FIELD_TYPE}
+   * e.g.Year:INTEGER:4,Make:STRING:5,Model:STRING:40,Description:STRING:40,
+   * Price:DOUBLE:8,Date:DATE:10:\"dd:mm:yyyy\" Date format needs to be within
+   * quotes (" ")
+   * 
+   * @param fieldDescription
+   */
+  public void setFieldDescription(String fieldDescription)
+  {
+    this.fieldDescription = fieldDescription;
+  }
+
+  /**
+   * Gets the character used for padding in the fixed-width file.Default is
+   * white space (' ')
+   * 
+   * @return
+   */
+  public char getPadding()
+  {
+    return padding;
+  }
+
+  /**
+   * Sets the character used for padding in fixed-width file.Default is white
+   * space (' ')
+   * 
+   * @param padding
+   */
+  public void setPadding(char padding)
+  {
+    this.padding = padding;
+  }
+
+  /**
+   * Extracts the fields from a fixed width record and returns a map containing
+   * field names and values
+   */
+  @Override
+  Map<String, Object> extractFields(String line)
+  {
+    if (!initialized) {
+      init();
+      initialized = true;
+    }
+    String[] values = fixedWidthParser.parseLine(line);
+    if (hasHeader && Arrays.deepEquals(values, header)) {
+      return null;
+    }
+    Map<String, Object> map = Maps.newHashMap();
+    int i = 0;
+    for (Field field : fields) {
+      map.put(field.getName(), getValue(field, values[i++]));
+    }
+    return map;
+  }
+
+  private void init()
+  {
+    fields = new ArrayList<Field>();
+    List<String> headers = new ArrayList<String>();
+    List<Integer> fieldWidth = new ArrayList<Integer>();
+    for (String tmp : fieldDescription.split(",")) {
+      String[] fieldTuple = tmp.split(":(?=([^\"]*\"[^\"]*\")*[^\"]*$)", -1);
+      Field field = new Field();
```

Comment:
Just to be moduler, can you create a new subclass called FixedWidthField which extends from AbstractCsvParser.Field and add additional fields like, field length and optional date format.
This way, all the information about the field is present in single structure.